### ISSUE TITLE
Accessibility violation in multiselect 

### ISSUE THREAD

**Issue by chriscarex** [jquense/react-widgets#1012]
Hi,
I'm currently using a multiselect in a project which has strict accessibility requirements.
When testing the component, I got the following errors:

1) 7 violations related to "Certain ARIA roles must contain particular children"
**Element location**: input[aria-owns="rw_7_listbox\ rw_7_notify_area\ "]
**Element source**:
<input tabindex="0" role="listbox" aria-expanded="false" aria-busy="false" aria-owns="rw_7_listbox rw_7_notify_area " aria-haspopup="true" placeholder="Filter by..." size="13" class="rw-input-reset" autocomplete="off" aria-disabled="false" aria-readonly="false" value="">

To solve this violation, you need to Fix the following:
Required ARIA child role not present: option

2) 14 violations related to "Buttons must have discernible text"
**Element location**: th:nth-child(4) > div > button:nth-child(2)
**Element source**: <button type="button">

To solve this violation, you need to Fix at least one (1) of these issues:
- Element does not have inner text that is visible to screen readers
- aria-label attribute does not exist or is empty
- aria-labelledby attribute does not exist, references elements that do not exist or references elements that are empty
- Element's default semantics were not overridden with role="presentation"
- Element's default semantics were not overridden with role="none"
- Element has no title attribute or the title attribute is empty

3) 7 violations related to "Form elements must have labels"
**Element location**: input[aria-owns="rw_7_listbox\ rw_7_notify_area\ "]
**Element source**:
<input tabindex="0" role="listbox" aria-expanded="false" aria-busy="false" aria-owns="rw_7_listbox rw_7_notify_area " aria-haspopup="true" placeholder="Filter by..." size="13" class="rw-input-reset" autocomplete="off" aria-disabled="false" aria-readonly="false" value="">

To solve this violation, you need to Fix at least one (1) of these issues:
- aria-label attribute does not exist or is empty
- aria-labelledby attribute does not exist, references elements that do not exist or references elements that are empty
- Form element does not have an implicit (wrapped) <label>
- Form element does not have an explicit <label>
- Element has no title attribute or the title attribute is empty

4) 7 best practice issues: "ARIA role must be appropriate for the element"
**Element location**: input[aria-owns="rw_7_listbox\ rw_7_notify_area\ "]
**Element source**:
<input tabindex="0" role="listbox" aria-expanded="false" aria-busy="false" aria-owns="rw_7_listbox rw_7_notify_area " aria-haspopup="true" placeholder="Filter by..." size="13" class="rw-input-reset" autocomplete="off" aria-disabled="false" aria-readonly="false" value="">

To solve this violation, you need to Fix the following:
ARIA role listbox is not allowed for given element

Let me know if that's clear and happy to help if needed!
Thanks,
Christian

**Comment by jquense** [comment]
> Let me know if that's clear and happy to help if needed!

Its not really. If you could synthesize the specific things you think are wrong and where from your tool's output that would help out, as i'm not familiar with the tool or how it describes violations

**Comment by chriscarex** [comment]
sure, I think that to fix the issues you should:
1) In the .rw-widget-input input, you should replace role="list-box" with role="option" and add aria-label="something"

2) add a title="something" or aria-label="something" prop to .rw-select button

Let me know if that's better now, you can test it easily analyzing it with the Axe chrome extension
